src/Sfa.Tl.ResultsAndCertification.Functions/CoreRommExtract.cs

```diff
@@ -0,0 +1,78 @@
+﻿using Microsoft.Azure.WebJobs;
+using Microsoft.Extensions.Logging;
+using Sfa.Tl.ResultsAndCertification.Application.Interfaces;
+using Sfa.Tl.ResultsAndCertification.Common.Enum;
+using Sfa.Tl.ResultsAndCertification.Common.Helpers;
+using Sfa.Tl.ResultsAndCertification.Functions.Helpers;
+using Sfa.Tl.ResultsAndCertification.Functions.Interfaces;
+using Sfa.Tl.ResultsAndCertification.Models.Configuration;
+using System;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Sfa.Tl.ResultsAndCertification.Functions
+{
+    public class CoreRommExtract
+    {
+        private readonly ICoreRommExtractService _coreRommExtractService;
+        private readonly ICommonService _commonService;
+        private readonly CoreRommExtractSettings _configuration;
+
+        public CoreRommExtract(
+            ResultsAndCertificationConfiguration configuration,
+            ICoreRommExtractService coreRommExtractService,
+            ICommonService commonService)
+        {
+            _configuration = configuration.CoreRommExtractSettings;
+            _coreRommExtractService = coreRommExtractService;
+            _commonService = commonService;
+        }
+
+        [FunctionName(Constants.CoreRomm)]
+        public async Task CoreRommExtractAsync([TimerTrigger("%CoreRommExtracTrigger%")] TimerInfo timer, ExecutionContext context, ILogger logger)
+        {
+            if (timer == null) throw new ArgumentNullException(nameof(timer));
+
+            var today = DateTime.UtcNow.Date;
+            bool shouldFunctionRunToday = _configuration.CoreRommValidDateRanges.Any(r => r.Contains(today));
+
+            if (!shouldFunctionRunToday)
+            {
+                await Task.CompletedTask;
+                return;
+            }
+
+            var functionLogDetails = CommonHelper.CreateFunctionLogRequest(context.FunctionName, FunctionType.AnalystCoreResultExtract);
```

Comment:
Please can we create new function type as this is different from AnalystCoreResultExtract.

Reply:
Corrected now